src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -14,39 +14,47 @@
 from pants.backend.python.rules.pex_from_target_closure import CreatePexFromTargetClosure
 from pants.backend.python.rules.prepare_chrooted_python_sources import ChrootedPythonSources
 from pants.backend.python.rules.pytest_coverage import (
+    COVERAGE_PLUGIN_INPUT,
     Coveragerc,
     CoveragercRequest,
     PytestCoverageData,
-    get_coverage_plugin_input,
 )
-from pants.backend.python.rules.targets import Coverage, Timeout
+from pants.backend.python.rules.targets import Coverage, PythonTestsSources, Timeout
 from pants.backend.python.subsystems.pytest import PyTest
 from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
 from pants.engine.addressable import Addresses
 from pants.engine.fs import Digest, DirectoriesToMerge, InputFilesContent
 from pants.engine.interactive_runner import InteractiveProcessRequest
 from pants.engine.isolated_process import ExecuteProcessRequest, FallibleExecuteProcessResult
 from pants.engine.legacy.graph import HydratedTargets, TransitiveHydratedTargets
-from pants.engine.legacy.structs import (
-    FilesAdaptor,
-    PythonTargetAdaptor,
-    PythonTestsAdaptorWithOrigin,
-    ResourcesAdaptor,
-    TargetAdaptorWithOrigin,
-)
+from pants.engine.legacy.structs import FilesAdaptor, PythonTargetAdaptor, ResourcesAdaptor
 from pants.engine.rules import UnionRule, rule, subsystem_rule
 from pants.engine.selectors import Get, MultiGet
+from pants.engine.target import TargetWithOrigin
 from pants.option.global_options import GlobalOptions
 from pants.python.python_setup import PythonSetup
-from pants.rules.core.determine_source_files import LegacySpecifiedSourceFilesRequest, SourceFiles
-from pants.rules.core.test import TestDebugRequest, TestOptions, TestResult, TestRunner
+from pants.rules.core.determine_source_files import SourceFiles, SpecifiedSourceFilesRequest
+from pants.rules.core.test import TestDebugRequest, TestOptions, TestResult, TestTarget
 
 
 @dataclass(frozen=True)
-class PytestRunner(TestRunner):
-    @staticmethod
-    def is_valid_target(adaptor_with_origin: TargetAdaptorWithOrigin) -> bool:
-        return isinstance(adaptor_with_origin, PythonTestsAdaptorWithOrigin)
+class PytestTarget(TestTarget):
```

Comment:
Re-using target sets the stage for heaps of confusion. How about `TestConfiguration` and `PythonTestConfiguration` (not a `PytestConfiguration` since the actual configuration fields here have nothing to do with pytest, they could work with nose).

The name comes from the original v2 experiment: https://github.com/pantsbuild/pants/commit/a71d25b9d513f06b6c00126127a360784a1baa02

Reply:
I like the name of `Configuration` to represent the idea of an ad hoc target. Thanks for the suggestion!

It's certainly better than my original `Fields` and better than the prior rule-specific name like `TestRunner` vs. `BinaryImplementation` vs. `Linter`.

Reply:
Excellent. There is probably more to look at from the original experiment when things get arounf to the BUILD side of this api.